Add definitions for RFC9218 (HTTP extensible prioritizations) (#312)
quicwg/qlog: draft-ietf-quic-qlog-h3-events.md

~~~
{: #h3-streamtypeset-def title="H3StreamTypeSet definition"}

## priority_updated {#h3-priorityupdated}
Importance: Base

> Emitted when a stream or server push's priority is initialized or updated

Reviewer: ```suggestion
Emitted when the priority of a request stream or push stream is initialized or updated
```

through mechanisms defined in {{!RFC9218}}. Specifically: an initial value is
set by default, an initial value is set through an HTTP Field in a HEADERS or
PUSH_PROMISE frame, or an updated value is set through an HTTP Field in a

> PRIORITY_UPDATE frame.

Author: Discussed on call: mainly be less prescriptive about the options (not Specifically but For instance). 

> For example, set initial prio, or when prio gets updated after HEADERS or PRIORITY_UPDATE, which can happen both due to signals from/decisions taken by client and server.

> PRIORITY_UPDATE frame.

Author: This was updated in the last commit. Please check again @LPardue  :)

Definition:

~~~ cddl
H3PriorityUpdated = {

>     ; if the prioritized element is a stream

Reviewer: ```suggestion
    ; if the prioritized element is a request stream
```

    ? stream_id: uint64

>     ; if the prioritized element is a push operation

Reviewer: ```suggestion
    ; if the prioritized element is a push stream
```

Author: Discussed during call: rename to `request stream` and `push stream` consistently

~~~
{: #h3maxpushidframe-def title="H3MaxPushIDFrame definition"}

### H3PriorityUpdateFrame

The PRIORITY_UPDATE frame is defined in {{!RFC9218}}.

~~~ cddl
H3PriorityUpdateFrame = {
    frame_type: "priority_update"

>     ; if the prioritized element is a stream

Reviewer: ```suggestion
    ; if the prioritized element is a request stream
```

    ? stream_id: uint64

>     ; if the prioritized element is a push operation

Reviewer: ```suggestion
    ; if the prioritized element is a push stream
```
